cookbooks/bcpc/recipes/chef_vault_install.rb

```diff
@@ -0,0 +1,38 @@
+#
+# Cookbook Name:: bcpc
+# Recipe:: chef vault install
+#
+# Copyright 2015, Bloomberg Finance L.P.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'pathname'
+require 'rubygems'
+
+gem_path = Pathname.new(Gem.ruby).dirname.join("gem").to_s
+chef_gemspec = "/opt/chef/embedded/lib/ruby/gems/1.9.1/specifications/chef-vault-2.2.4.gemspec"
+
+gem_package "chef-vault" do
+  gem_binary gem_path
+  version ">=0.0.0"
+  action :nothing
```

Comment:
Let's see if you can set a `umask` attribute on a `gem_package` resource to work around the restrictive [knife.rb](https://github.com/bloomberg/chef-bach/blob/5c07b64905beb52b32e876b0ac8a5f2a0de14b95/setup_chef_cookbooks.sh#L55) umask. I would like that more than a file action to simply change the permissions; only I suspect umask is not going to be supported.

Otherwise, if the `gem_package` does not support umask, could you add a comment on line 34 that the file resource is only to set the permissions? Further, it seems the `Gem.clear_paths` should be under the file action as that is really only when the gem is "deployed" -- though here we are running as root so get away with it before the file action.